#! /usr/bin/env python3
PACKAGE = "sub8_perception"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
# using strings because there seems to be no vector support (currently)
gen.add("target", str_t, 0, "Target Vampire", "drac")
gen.add("override", bool_t, 0, "Override Vampire Default Values?", False)
gen.add("dyn_lower", str_t, 0, "Lower RGB Threshold", "0,0,80")
gen.add("dyn_upper", str_t, 0, "Upper RGB Threshold", "200, 200, 250")
gen.add(
    "min_trans",
    double_t,
    0,
    "Minimum Transformation for Observation (meters)",
    0.05,
    0,
    2,
)
gen.add("max_velocity", int_t, 0, "Maximum Velocity for Observation", 1)
gen.add(
    "timeout", int_t, 0, "Maximum time until an Observation expires", 25000, 0, 50000
)
gen.add("min_obs", int_t, 0, "Minimum Observations before Point Estimation", 10, 4, 16)
exit(gen.generate(PACKAGE, PACKAGE, "VampireIdentifier"))
